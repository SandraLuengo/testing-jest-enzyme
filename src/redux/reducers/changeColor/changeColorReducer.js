import initialState from './initialState';

export default (state = initialState, action) => {
  switch (action.type) {
    case 'TOGGLE_COLOR':
      return {
        toggleColor: action.payload,
      };
    default:
      return state;
  }
};
