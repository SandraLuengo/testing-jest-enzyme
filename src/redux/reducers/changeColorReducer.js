import initialState from './initialState';

export default (state = initialState, action) => {
  switch (action.type) {
    case 'TOGGLE_COLOR':
      return {
        toggleColor: action.newValue,
      };
    default:
      return state;
  }
};
