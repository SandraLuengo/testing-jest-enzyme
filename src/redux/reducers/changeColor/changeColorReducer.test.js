import changeColorReducer from './changeColorReducer';

describe('TOGGLE_COLOR', () => {
  it('initialState is correct', () => {
    const action = { type: 'TESTING_TOGGLE_COLOR' };
    const initialState = { toggleColor: false };
    // expect(changeColorReducer(undefined, action)).toEqual(initialState);
    expect(changeColorReducer(undefined, action).contact).toMatchSnapshot();
  });
  it('returns the correct state', () => {
    const action = { type: 'TOGGLE_COLOR', payload: true };
    const expectedState = { toggleColor: true };
    // expect(changeColorReducer(undefined, action)).toEqual(expectedState);
    expect(changeColorReducer(undefined, action)).toMatchSnapshot();
  });
});

