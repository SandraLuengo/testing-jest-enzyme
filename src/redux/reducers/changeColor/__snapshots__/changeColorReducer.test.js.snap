// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOGGLE_COLOR initialState is correct 1`] = `undefined`;

exports[`TOGGLE_COLOR returns the correct state 1`] = `
Object {
  "toggleColor": true,
}
`;
